src/daft-sql/src/functions.rs

```diff
@@ -12,27 +13,357 @@ use crate::{
+    /// # Example
+    /// ```sql
+    /// SELECT starts_with('hello', 'he');
+    /// ```
+    StartsWith,
+    /// SQL `contains()` function
+    /// # Example
+    /// ```sql
+    /// SELECT contains('hello', 'el');
+    /// ```
+    Contains,
+    /// SQL `split()` function
+    /// # Example
+    /// ```sql
+    /// SELECT split('hello,world', ',');
+    /// ```
+    Split,
+
+    Extract,
+    ExtractAll,
+    /// SQL 'replace()' function
+    /// # Example
+    /// ```sql
+    /// SELECT replace('hello', 'l', 'r');
+    /// ```
+    Replace,
+    /// SQL 'length()' function
+    /// # Example
+    /// ```sql
+    /// SELECT length('hello');
+    /// ```
+    Length,
+    /// SQL 'lower()' function
+    /// # Example
+    /// ```sql
+    /// SELECT lower('HELLO');
+    /// ```
+    Lower,
+    /// SQL 'upper()' function
+    /// # Example
+    /// ```sql
+    /// SELECT upper('hello');
+    /// ```
+    Upper,
+    /// SQL 'lstrip()' function
+    /// # Example
+    /// ```sql
+    /// SELECT lstrip(' hello');
+    /// ```
+    Lstrip,
+    /// SQL 'rstrip()' function
+    /// # Example
+    /// ```sql
+    /// SELECT rstrip('hello ');
+    /// ```
+    Rstrip,
+    /// SQL 'reverse()' function
+    /// # Example
+    /// ```sql
+    /// SELECT reverse('olleh');
+    /// ```
+    Reverse,
+    /// SQL 'capitalize()' function
+    /// # Example
+    /// ```sql
+    /// SELECT capitalize('hello');
+    /// ```
+    Capitalize,
+    /// SQL 'left()' function
+    /// # Example
+    /// ```sql
+    /// SELECT left('hello', 2);
+    /// ```
+    Left,
+
+    /// SQL 'right()' function
+    /// # Example
+    /// ```sql
+    /// SELECT right('hello', 2);
+    /// ```
+    Right,
+    /// SQL 'find()' function
+    /// # Example
+    /// ```sql
+    /// SELECT find('hello', 'l');
+    /// ```
+    Find,
+    /// SQL 'rpad()' function
+    /// # Example
+    /// ```sql
+    /// SELECT rpad('hello', 10, ' ');
+    /// ```
+    Rpad,
+    /// SQL 'lpad()' function
+    /// # Example
+    /// ```sql
+    /// SELECT lpad('hello', 10, ' ');
+    /// ```
+    Lpad,
+    /// SQL 'repeat()' function
+    /// # Example
+    /// ```sql
+    /// SELECT repeat('X', 2);
+    /// ```
+    Repeat,
+    // Like,
+    // Ilike,
+    /// SQL 'substring()' function
+    /// # Example
+    /// ```sql
+    /// SELECT substring('hello', 1, 2);
+    /// ```
+    Substr,
+    /// SQL 'to_date()` function
+    /// # Example
+    /// ```sql
+    /// SELECT to_date('2021-01-01', 'YYYY-MM-DD');
+    /// ```
+    ToDate,
+    /// SQL 'to_datetime()' function
+    /// # Example
+    /// ```sql
+    /// SELECT to_datetime('2021-01-01 00:00:00', 'YYYY-MM-DD HH:MM:SS');
+    /// ```
+    ToDatetime,
+
+    // ------------------------------------------------
+    // Aggregate Functions
+    // ------------------------------------------------
     Max,
 }
 
 impl FromStr for SQLFunctions {
     type Err = PlannerError;
 
     fn from_str(value: &str) -> Result<Self, Self::Err> {
```

Comment:
If you are tired of writing all the match arms by hand we could use [strum](https://docs.rs/strum_macros/latest/strum_macros/derive.EnumString.html) to auto-generate this!